### PR TITLE
PubsubImprovement

### PR DESCRIPTION
**Pull request by altayhunoglu**

### CODE REVIEW

**Review by ErdemT09 — APPROVED** [review]

File: src/common/websocket/middleware.js [hunk 1]
```diff
@@ -1,113 +1,149 @@
...
+      case 'LOGOUT':
+        if (clientNew) {
+          clientNew.close();
+          console.log('websocket closed');
         }
-    };
+        return next(action);
+          
+      case 'WS_DISCONNECT':
+        console.log('websocket closed');
```
Comment (ErdemT09) [inline src/common/websocket/middleware.js:137]: Shouldn't we also close when we disconnect?

Comment (altayhunoglu) [reply src/common/websocket/middleware.js:137]: Absolutely. Nice catch. I will fix it.

**Review by ErdemT09 — APPROVED** [review]
The middleware contains the essential part of the websocket system. Each channel type, transaction type, however we may call it, request its own stream. Each requested stream is accessed by the same socket variable though.

File: src/common/websocket/middleware.js [hunk 2]
```diff
@@ -52,9 +54,10 @@ const socketMiddleware = () => {
   }
   const notificationConnectionRequestChannel = (username, store) => {
     if(socket==null) return;
+    var unique
```
Comment (ErdemT09) [inline src/common/websocket/middleware.js:57]: `  var uniqueIdForTheClient = uuidv4();
`
For this too.

**Pull request merged**